src/main/scala/is/hail/stats/RegressionUtils.scala

```diff
@@ -185,6 +233,27 @@ object RegressionUtils {
       None
   }
 
+  def parseXExpr(xExpr: String, ec: EvalContext): () => java.lang.Double = {
+    val (xt, xf0) = Parser.parseExpr(xExpr, ec)
+
+    def castToDouble[T](f: (T) => Double): () => java.lang.Double = { () =>
```

Comment:
There's duplication between the logic here and the logic in `toDouble` and `getSampleAnnotation[s]` just below where toDouble is used, so consider pulling that out.

There's another duplication is logically separate from this PR: getSampleAnnotation is a special case of getSampleAnnotations and getPhenoCovCompleteSamples is a special case of getPhenosCovCompleteSamples. I left the former in when creating the latter a long time ago so as not to add extra nesting overhead. Do you think I should remove them?

Reply:
Probably be fine to remove them, but I might leave it for when you go back to the generalized regression stuff.